docs/decisions/07-posts-data-model.md

```diff
@@ -0,0 +1,162 @@
+    Post --> FeedDto : maps-to
+    FeedDto --* Content : contains
+```
+
+## Impact
+
+This will obviously introduce some new data types. It's also likely to require some complex migrations. It should ideally happen soon, before any serious work starts on posting-related features.
+
+You may also notice that this data model is more restrictive than what ActivityPub would allow. Notably, Posts can only be InReplyTo at most one other Post (and no other kind of object). A Note can only be in one Post and vice-versa. And because `IContent.Thread` would map to `AS Context`, a post can only have one context and exist in one thread. These restrictions should be fully compatible with the actual behavior of other fedi services in the wild. It helps to narrow the possibility space that we would actually have to deal with, and should support faster queries for the most common cases (when a post is a note and nothing else). This means that there could be potential future fedi peers that we would not be readily compatible with.
+
+## Context
+
+The content-related data schema we have now is troublesome to work with, and that will only get worse over time. It was developed quickly, very early in the project, and it was always likely that it wouldn't be viable in the long term. The original schema was useful to provide some anchor points to develop an internal data model that isn't simply replicating all of ActivityPub (a likely impossible task). It also provided a point to iterate from. Now we're iterating, with more care and more context.
+
+For reference, the model we have now is that various classes (`Note` and `Image` at the moment) implement an `IContentRef` interface, and that's kind of it. Those objects can be assigned a resolvable URI based on their types, and those URIs are the critical component of Feed records. To actually use this system in practice, the client would retrieve a batch of records for the feed, and it would consist of little more than URI which it would have to then issue individual follow-up requests to resolve. Which is not great. It's also surprisingly difficult to map AP docs to these objects. I'm not sure this design solves that problem on its own, but I think it does become easier to build a `ContentFactory` or similar, which was likely always going to be necessary. Then we can map them to `Content` objects, as built by the factory, which can handle the logic of deciding what concrete type to construct.
+
+## Discussion
+
+### minor activities
+A Post can have more than one content, and a content item can theoretically exist in more than one post. If a post is `Liked` by someone, should that Like follow the content to other posts? I think probably not, right? For example, suppose you upload a photo, then post that photo with a note. The later you add that photo to an album. Should the view of the photo in the album show all the Likes that the Post received? What if you reuse the photo in another post in the future?
```

Comment:
```suggestion
### minor activities
A Post can have more than one content, and a content item can theoretically exist in more than one post. If a post is `Liked` by someone, should that Like follow the content to other posts? I think probably not, right? For example, suppose you upload a photo, then post that photo with a note. The later you add that photo to an album. Should the view of the photo in the album show all the Likes that the Post received? What if you reuse the photo in another post in the future?
```

Ignore this, it's only relevant to an earlier iteration that I think was too complex and might have presented performance concerns